internal/server/server.go

```diff
@@ -155,17 +161,27 @@ func (s *Server) ConfigureOpenID() error {
 	return nil
 }
 
-func (s *Server) routes() {
-	s.HandleFunc("/", s.handleRoot())
-	s.HandleFunc("/login", s.handleLogin()).Methods(http.MethodPost)
-	s.HandleFunc("/callback", s.handleCallback()).Methods(http.MethodGet)
-}
-
 func logAndError(w http.ResponseWriter, code int, err error, msg string) {
 	log.WithError(err).Error(msg)
 	http.Error(w, http.StatusText(code), code)
 }
 
+func writeJsonResponse(w http.ResponseWriter, httpResponse int, data interface{}) {
+	jsonResp, err := json.Marshal(data)
+	if err != nil {
+		logAndError(w, http.StatusInternalServerError, err, "error marshaling json")
+		return
```

Comment:
If this error condition, nothing is written to the `ResponseWriter` which will cause the connection to hang.

Reply:
Hmm I think the logAndError -> http.Error() call will write to the `ResponseWriter` correctly with error and code.